pkg/webui/console/views/device-general-settings/identity-server-form/validation-schema.js

```diff
@@ -0,0 +1,92 @@
+// Copyright © 2019 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import * as Yup from 'yup'
+
+import randomByteString from '../../../lib/random-bytes'
+import sharedMessages from '../../../../lib/shared-messages'
+import { id as deviceIdRegexp, address as addressRegexp } from '../../../lib/regexp'
+import m from '../../../components/device-data-form/messages'
+
+import { parseLorawanMacVersion } from '../utils'
+
+const random16BytesString = () => randomByteString(32)
+const toUndefined = value => (!Boolean(value) ? undefined : value)
```

Comment:
We use this frequently, so I'd say it would be better to outsource into a module.